### DISCUSSION TITLE
How to create new card and read it by CFG file?

### DISCUSSION THREAD

**Discussion by halfwitbm** [OpenRadioss/OpenRadioss#1675] [Q&A] [answered]
Hi everyone,

I tried to add a new card variable and read it by CFG file as OpenRadioss help listed: https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/6094849/OpenRadioss+Reader+Radioss+Block+Format#.cfg-Files

I did it as following steps:
Step 1: added new card information on workable input file (as help showing)
Step 2: created a new folder named NEWTYPE1 at hm_cfg_files/config/CFG/radioss2022/, then created a file named as newtype1.cfg, and it content as help showing.
Step 3: added some new data hierarchy information as help showing at the end of data_hierarchy.cfg in hm_cfg_files/config/CFG/radioss2022/
Step 4: created a folder named it as “newtype1” at starter/source, created hm_read_new_type1.F in this folder and its content as that in help.
Step 5: Call HM_OPTION_COUNT to read this new card in contrl.F (count by Calling HM_OPTION_COUNT) or call hm_read_new_type1.F in the lectur.F

I follow all steps in help file, but this new card content still can not be read. When counting other known card such as "/NODE", the subroutine HM_OPTION_COUNT works well.

Had anybody else ever test these steps to read new card? Did this kind of new card reading method work?
Could you please share the experience with me? Thank you very much.

Bin

**Comment by sebastienVilleneuve** [comment, marked as answer]
Hi @halfwitbm 

can, you put the modifications you've done in  hm_cfg_files/config/CFG/radioss2023/ , and retest please ?

(You need to modify data_hierarchy.cfg file of the last available directory for OpenRadioss input  )
For sure , picture is not up to date , I need to update this


Thanks,
Sébastien

**Comment by halfwitbm** [comment]
Hello Sébastien,

Thank you very much for the solution. I tested your suggestion. It worked
perfectly.

Thanks again for your help.

Best,

Bin
